update-db.js

```diff
@@ -180,17 +180,31 @@ module.exports = function updateDB (print) {
   var latest = getLatestInfo(lock)
   var browsersListRetrievalError
   var oldBrowsersList
+  var currentIsLatest = false
   try {
     oldBrowsersList = getBrowsersList()
   } catch (e) {
     browsersListRetrievalError = e
   }
 
   if (typeof current === 'string') {
+    if (current === latest.version) {
```

Comment:
What is `yarn.lock` contains different versions of `caniuse-lite`?

Reply:
Do you mean that you want a test for yarn, too?

Reply:
Nope.

Both `yarn.lock` and `package-lock.json` can contains a different versions of `caniuse-lite` (because of dublication problem of npm).

`--update-db` not only updates `caniuse-lite` version, but also remove dublicates.

So we need also to check that there is not different `caniuse-lite` versions in lock file.

Reply:
It's an existing problem, then -- `getCurrentVersion` reports the first version of `caniuse-lite` that it comes across, rather than checking to see if multiple versions are installed in the various dependencies, and if so, what the version numbers are. There's also nothing in the test fixtures to ensure that `update-db.js` handles the duplicate situation correctly.

Reply:
> It's an existing problem, then -- getCurrentVersion reports the first version of caniuse-lite that it comes across

Yeap. But nod duplication will be fixed anyway (even if the bigger version from the lock file is the latest). This bug affects only CLU output.

After this PR `--update-db` will not fix `caniuse-lite` duplicates if the bigger version from the lock file is the latest.

Reply:
How about if there are multiple versions of `caniuse-lite` because `package.json` specifies a particular version? The current search and replace approach takes no account of this (edge) case.

Reply:
In the current approach, we remove all `caniuse-lite` mentions from the lock and force lock file update. `npm` or `yarn` will download the latest `caniuse-lite` for dependencies requirements.

So if the app locked the version, it will have the locked version. But other tools will have the latest since they didn’t lock the version. It is an expected approach since `package.json` is not a way to lock dependency for the whole dependencies tree.

Reply:
I suggest improving the check. We should find the smallest `caniuse-lite` versions in the lock file and run the update script if the smallest is not equal to the latest.

Reply:
I've rewritten `update-db.js`to find all versions of `caniuse-lite` in the lock file, and rewrite the file and trigger an update if there are any versions other than the latest present.